ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/search_tree/algorithms/SearchTree.java

```diff
@@ -482,4 +494,46 @@ private AppliedRemedialActions getAppliedRemedialActions(RangeActionActivationRe
         }
         return alreadyAppliedRa;
     }
+
+    /**
+     * If stop criterion is reached on functional cost, but not with virtual cost, this method logs information
+     * about postivie virtual costs
+     */
+    private void logVirtualCostInformation(Leaf leaf, String prefix) {
+        leaf.getVirtualCostNames().stream()
+                .filter(virtualCostName -> leaf.getVirtualCost(virtualCostName) > 1e-6)
+                .forEach(virtualCostName -> logVirtualCostDetails(leaf, virtualCostName, prefix));
+    }
+
+    void logVirtualCostDetails(Leaf leaf, String virtualCostName, String prefix) {
+        FaraoLogger logger = topLevelLogger;
+        if (!costSatisfiesStopCriterion(leaf.getCost())
+                && costSatisfiesStopCriterion(leaf.getCost() - leaf.getVirtualCost(virtualCostName))
```

Comment:
not really, because here i am only removing the specific virtual cost for "virtualCostNAme", in order to detect if it is really this one that is causing trouble (and not the sum of all virtual costs)